ledger-core/v2/virtual/integration/method_test.go

```diff
@@ -336,3 +336,60 @@ func TestVirtual_Method_WithExecutor(t *testing.T) {
 		<-testIsDone
 	}
 }
+
+func TestVirtual_CallMethodAfterPulseChange(t *testing.T) {
+	server := utils.NewServer(t)
+	ctx := inslogger.TestContext(t)
+
+	server.IncrementPulse(ctx)
+
+	var (
+		vCallRequestCount  uint
+		vStateRequestCount uint
+		vStateReportCount  uint
+		vCallResultCount   uint
+	)
+
+	server.PublisherMock.Checker = func(topic string, messages ...*message.Message) error {
+		require.Len(t, messages, 1)
+
+		pl, err := payload.UnmarshalFromMeta(messages[0].Payload)
+		require.NoError(t, err)
+
+		switch pl.(type) {
+		case *payload.VCallResult:
+			vCallResultCount++
+		case *payload.VCallRequest:
+			vCallRequestCount++
+		case *payload.VStateReport:
+			vStateRequestCount++
+		case *payload.VStateRequest:
+			vStateReportCount++
+		default:
+			require.Failf(t, "", "bad payload type, expected %s, got %T", "*payload.VCallResult", pl)
+		}
+
+		server.SendMessage(ctx, messages[0])
+		return nil
+	}
+
+	testBalance := uint32(555)
+	rawWalletState := makeRawWalletState(t, testBalance)
+	objectRef := reference.NewRecordRef(server.RandomLocalWithPulse())
+	{
+		// send VStateReport: save wallet
+		msg := makeVStateReportEvent(t, objectRef, rawWalletState)
+		require.NoError(t, server.AddInput(ctx, msg))
+	}
+
+	// Change pulse to force send VStateRequest
+	server.IncrementPulse(ctx)
```

Comment:
why do you need to change pulse? Do we already have migrations and stuff?

Reply:
I change pulse since I want to check that we really send VStateRequest to previous executer